lib/improver/argparser.py

```diff
@@ -185,6 +178,14 @@ def add_arguments(self, argspec_list):
             argflags, argkwargs = argspec
             self.add_argument(*argflags, **argkwargs)
 
+    def parse_args(self, args=None, namespace=None):
+        """Wrap in order to implement some compulsory behaviour."""
+        args = super(ArgParser, self).parse_args(args=args,
+                                                 namespace=namespace)
+        if hasattr(args, 'profile') and (args.profile or args.profile_file):
```

Comment:
I *think* this can just be:
`if args.profile or args.profile_file`?, assuming that the compulsory options were actually added (i.e: if they were added, `hasattr(args, 'profile')` should always be `True`?)

Reply:
(But I've no objection to the extra check)

Reply:
Ah, this first check is a weird thing I had to add for the test_argparser unit tests to work. You are right otherwise